### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by Nahid1911**
### REQUIREMENTS:

Please click [HERE](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_join_table.md) for details requirements.

### Answer of the queries:

1. Who was the last animal seen by William Tatcher?
image

2. How many different animals did Stephanie Mendez see?
image

3. List all vets and their specialties, including vets with no specialties.
image

4. List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
image

5. What animal has the most visits to vets?
image

6. Who was Maisy Smith's first visit?
image

7. Details for most recent visit: animal information, vet information, and date of visit.
image

8. How many visits were with a vet that did not specialize in that animal's species?
image

9. What specialty should Maisy Smith consider getting? Look for the species she gets the most.
image

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -32,3 +32,23 @@ DROP species;
...
+
+CREATE TABLE specializations (
+species_id INT REFERENCES species(id),
+vets_id INT REFERENCES vets(id)
+);
+
+CREATE TABLE visits (
+animals_id INT REFERENCES animals(id),
+vets_id INT REFERENCES vets(id)
+);
```
Comment (mikethreels) [inline schema.sql]: - [ ] in the visits table there should be a column with date included as well

File: queries.sql [hunk 2]
```diff
@@ -98,4 +98,37 @@ SELECT a.name, a.escape_attempts, o.full_name FROM animals AS a INNER JOIN owner
...
+
+
+----------------------Day-4-Project--------------------------------
+
+-- Who was the last animal seen by William Tatcher?
+SELECT a.name AS last_visited_animal FROM visits JOIN vets ON visits.vets_id = vets.id JOIN animals AS a ON visits.animals_id = a.id WHERE vets.name = 'William Tatcher' ORDER BY visit_date DESC LIMIT 1;
+
+
+-- How many different animals did Stephanie Mendez see?
+SELECT COUNT(s.name) AS visit_different_animals, vets.name AS doctor_name FROM specializations JOIN vets ON specializations.vets_id = vets.id JOIN species AS s ON specializations.species_id = s.id WHERE vets.name = 'Stephanie Mendez' GROUP BY vets.name;
```
Comment (mikethreels) [inline queries.sql]: - [ ] something seems incorrect in this case looking at the data you put in the table, Stephanie Mendez saw 4 different animals, it seems you're returning how many species she saw

File: queries.sql [hunk 3]
```diff
@@ -98,4 +98,37 @@ SELECT a.name, a.escape_attempts, o.full_name FROM animals AS a INNER JOIN owner
...
+SELECT COUNT(a.name) AS ttl_visits, a.name AS anim_name FROM visits AS v JOIN animals AS a ON v.animals_id = a.id GROUP BY a.name ORDER BY ttl_visits DESC LIMIT 1 ;
+
+-- Who was Maisy Smith's first visit?
+SELECT MIN(visit_date) AS first_visit, a.name AS anim_name FROM visits AS v JOIN vets AS vt ON v.vets_id = vt.id JOIN animals AS a ON v.animals_id = a.id WHERE vt.name = 'Maisy Smith' GROUP BY a.name ORDER BY first_visit ASC LIMIT 1;
+
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT MAX(visit_date) AS recent_visit, a.name AS anim_name, vt.name AS vet_name FROM visits AS v JOIN animals AS a ON v.animals_id = a.id JOIN vets AS vt ON v.vets_id = vt.id GROUP BY a.name, vt.name ORDER BY recent_visit DESC LIMIT 1;
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(visit_date) AS num_visit_with_unspecialize_vet FROM specializations AS sp JOIN vets AS vt ON sp.vets_id = vt.id JOIN visits AS v ON sp.vets_id = v.vets_id WHERE species_id IS NULL;
```
Comment (mikethreels) [inline queries.sql:131]: - [ ] In this query you're listing the number of visits with a vet that doesn't have any specialization, however, we are looking for all the visits with a vet that doesn't specialize in that species which should add up to 12.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Nahid, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job writing the queries.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Nahid1911) [reply schema.sql]: @mikethreels Thank you for your review, the code has been corrected accordingly.

File: queries.sql (hunk 2 above)
Comment (Nahid1911) [reply queries.sql]: @mikethreels Thank you for your review, the code has been corrected accordingly and result is 4 now.

File: queries.sql (hunk 3 above)
Comment (Nahid1911) [reply queries.sql:131]: @mikethreels Thank you for your review, the code has been corrected accordingly and the result is 12 now.

**Review by topeogunleye — APPROVED** [review]
### Hi @Nahid1911, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Added date column to visits table. ✔️ 
- Fixed query for "* How many times visits where with a vet that did not specialize in that animal's species?" ✔️ 
- Fixed query for "How many different animals did Stephanie Mendez see?" ✔️ 
 
## [Optional] suggestions:
- For future pull requests, please notice that your branch name `joinTbl` does not follow the convention of being named after a feature or action. For example, you could name your branch `addedJoinTbl`. This makes it easier to track the changes and the purpose of each branch. 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**